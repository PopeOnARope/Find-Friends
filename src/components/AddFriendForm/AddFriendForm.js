import React, { Component } from "react";
import { connect } from "react-redux";
import { AddFriendButton } from "../../App.js";
import { addFriend } from "../../redux/reducers";

import { Collapse } from "react-collapse";

import glamorous, { ThemeProvider } from "glamorous";

const Label = glamorous.form({
  padding: "0.5rem",
  width: "100%",
  display: "flex"
});

export class AddFriendForm extends React.Component {
  constructor() {
    super();
    this.state = { name: "", avatar: "" };
  }

  updateFormValues(target) {
    this.setState({ [target.name]: target.value });
  }
  render() {
    const { showAddFriendForm, addFriend } = this.props;
    return (
      <Collapse isOpened={showAddFriendForm}>
        <form name="Add a Friend" style={{ padding: "0.5rem" }}>
          <legend>Fill in the information below</legend>
          <Label>
            Name:
            <input
              name="name"
              onChange={e => this.updateFormValues(e.target)}
              type="text"
            />
          </Label>
          <Label>
            Avatar:
            <input
              name="avatar"
              onChange={e => this.updateFormValues(e.target)}
              type="text"
            />
          </Label>
          <AddFriendButton
            className="button"
            onClick={e => {
              e && e.preventDefault();
              addFriend(this.state);
            }}
          >
            Add Friend
          </AddFriendButton>
        </form>
      </Collapse>
    );
  }
}

const ConnectedComponent = connect(state => state, { addFriend })(
  AddFriendForm
);
export default ConnectedComponent;
