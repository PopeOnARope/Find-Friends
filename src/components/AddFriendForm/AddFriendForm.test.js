import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { AddFriendForm } from "./AddFriendForm";

configure({ adapter: new Adapter() });

const mockProps = {
  user: {
    name: "John Pope",
    friends: [
      {
        avatar: "https://placekitten.com/g/25/26",
        name: "Jim",
        lat: 32.796421,
        lng: -79.9312164
      }
    ]
  },
  showAddFriendForm: false
};

it("Renders as expected", () => {
  const wrapper = shallow(<AddFriendForm {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("adds a new friend when the add friend button is clicked a second time", () => {
  const addFriend = jest.fn();
  const toggleAddFriendForm = jest.fn();
  const wrapper = shallow(
    <AddFriendForm
      {...mockProps}
      showAddFriendForm={true}
      toggleAddFriendForm={toggleAddFriendForm}
      addFriend={addFriend}
    />
  );
  wrapper.find(".button").simulate("click");
  expect(addFriend).toHaveBeenCalled();
});
