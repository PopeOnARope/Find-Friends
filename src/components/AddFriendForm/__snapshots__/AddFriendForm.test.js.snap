// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddFriendForm
    showAddFriendForm={false}
    user={
        Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
            "name": "John Pope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <form
        name="Add a Friend"
        style={
                Object {
                        "padding": "0.5rem",
                      }
        }
>
        <legend>
                Fill in the information below
        </legend>
        <glamorous(form)>
                Name:
                <input
                        name="name"
                        onChange={[Function]}
                        type="text"
                />
        </glamorous(form)>
        <glamorous(form)>
                Avatar:
                <input
                        name="avatar"
                        onChange={[Function]}
                        type="text"
                />
        </glamorous(form)>
        <glamorous(button)
                className="button"
                onClick={[Function]}
        >
                Add Friend
        </glamorous(button)>
</form>,
      "fixedHeight": -1,
      "forceInitialAnimation": false,
      "hasNestedCollapse": false,
      "isOpened": false,
      "onMeasure": [Function],
      "onRender": [Function],
      "onRest": [Function],
      "style": Object {},
      "theme": Object {
        "collapse": "ReactCollapse--collapse",
        "content": "ReactCollapse--content",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <legend>
            Fill in the information below
</legend>,
          <glamorous(form)>
            Name:
            <input
                        name="name"
                        onChange={[Function]}
                        type="text"
            />
</glamorous(form)>,
          <glamorous(form)>
            Avatar:
            <input
                        name="avatar"
                        onChange={[Function]}
                        type="text"
            />
</glamorous(form)>,
          <glamorous(button)
            className="button"
            onClick={[Function]}
>
            Add Friend
</glamorous(button)>,
        ],
        "name": "Add a Friend",
        "style": Object {
          "padding": "0.5rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fill in the information below",
          },
          "ref": null,
          "rendered": "Fill in the information below",
          "type": "legend",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Name:",
              <input
                name="name"
                onChange={[Function]}
                type="text"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Avatar:",
              <input
                name="avatar"
                onChange={[Function]}
                type="text"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Avatar:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "avatar",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add Friend",
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Friend",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <form
          name="Add a Friend"
          style={
                    Object {
                              "padding": "0.5rem",
                            }
          }
>
          <legend>
                    Fill in the information below
          </legend>
          <glamorous(form)>
                    Name:
                    <input
                              name="name"
                              onChange={[Function]}
                              type="text"
                    />
          </glamorous(form)>
          <glamorous(form)>
                    Avatar:
                    <input
                              name="avatar"
                              onChange={[Function]}
                              type="text"
                    />
          </glamorous(form)>
          <glamorous(button)
                    className="button"
                    onClick={[Function]}
          >
                    Add Friend
          </glamorous(button)>
</form>,
        "fixedHeight": -1,
        "forceInitialAnimation": false,
        "hasNestedCollapse": false,
        "isOpened": false,
        "onMeasure": [Function],
        "onRender": [Function],
        "onRest": [Function],
        "style": Object {},
        "theme": Object {
          "collapse": "ReactCollapse--collapse",
          "content": "ReactCollapse--content",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <legend>
              Fill in the information below
</legend>,
            <glamorous(form)>
              Name:
              <input
                            name="name"
                            onChange={[Function]}
                            type="text"
              />
</glamorous(form)>,
            <glamorous(form)>
              Avatar:
              <input
                            name="avatar"
                            onChange={[Function]}
                            type="text"
              />
</glamorous(form)>,
            <glamorous(button)
              className="button"
              onClick={[Function]}
>
              Add Friend
</glamorous(button)>,
          ],
          "name": "Add a Friend",
          "style": Object {
            "padding": "0.5rem",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fill in the information below",
            },
            "ref": null,
            "rendered": "Fill in the information below",
            "type": "legend",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Name:",
                <input
                  name="name"
                  onChange={[Function]}
                  type="text"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Name:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "name",
                  "onChange": [Function],
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Avatar:",
                <input
                  name="avatar"
                  onChange={[Function]}
                  type="text"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Avatar:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "avatar",
                  "onChange": [Function],
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Friend",
              "className": "button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add Friend",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
